zebra-state/src/service/finalized_state/zebra_db.rs

```diff
@@ -169,6 +178,42 @@ impl ZebraDb {
             if let Some(format_change_handle) = self.format_change_handle.as_mut() {
                 format_change_handle.force_cancel();
             }
+
+            // # Correctness
+            //
+            // Check that the database format is correct before shutting down.
+            // This lets users know to delete and re-sync their database immediately,
+            // rather than surprising them next time Zebra starts up.
+            //
+            // # Testinng
+            //
+            // In Zebra's CI, panicking here stops us writing invalid cached states,
+            // which would then get make unrelated PRs fail when Zebra starts up.
```

Comment:
###### Optional Suggestion

```suggestion
            // which would then make unrelated PRs fail when Zebra starts up.
```